src/ring_theory/dedekind_domain/integer_unit.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import ring_theory.dedekind_domain.adic_valuation
+
+/-!
+# `S`-integers and `S`-units of fraction fields of Dedekind domains
+
+This file defines the ring of `S`-integers and the group of `S`-units of the field of fractions of a
+Dedekind domain, to be specialised to the case of a number field or a function field separately.
+
+TODO:
+* notation for `S`-integers and `S`-units.
+* proof that `S`-integers is the intersection of valuation rings.
+* proof that `S`-units is the kernel of a map to a product.
+* proof that `∅`-integers is the usual ring of integers.
+* finite generation of `S`-units and Dirichlet's `S`-unit theorem.
```

Comment:
These doc strings are parsed as markdown so if you indent lists like this then the HTML rendered in our docs will be a proper list:
```suggestion
 * notation for `S`-integers and `S`-units.
 * proof that `S`-integers is the intersection of valuation rings.
 * proof that `S`-units is the kernel of a map to a product.
 * proof that `∅`-integers is the usual ring of integers.
 * finite generation of `S`-units and Dirichlet's `S`-unit theorem.
```